enhancements/machine-config/securing-mcs.md

```diff
@@ -0,0 +1,112 @@
+
+## Design Details
+
+### Open Questions
+
+1. Can we avoid the extra reboot when post-node-approved Machine Configs are used?
+
+### Test Plan
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+#### Removing a deprecated feature
+
+We aren't planning on following a deprecation cycle for this change. We could make use of CRD versioning and phase this in, but the underlying driver for this change is the removal of the TCP port restriction and that wouldn't happen until the deprecation is complete. This timeline would very likely be unacceptable to the customer.
+
+### Upgrade / Downgrade Strategy
+
+Existing machine configs are not modified, including rendered configs, so downgrades will be straightforward (for the Machine Config Operator), though they will leak rendered Machine Configs. The Machine Configs that are created by the Machine Config Operator would be placed into the pre-node-approved set, while any customer-provided Machine Configs would be placed into the post-node-approved set. This ensures that no configuration is inadvertently exposed as the result of an upgrade.
+
+This runs the risk of breaking auto-scaling because certain Machine Configs are not included in the generated Ignition config. Short of selectively-enforcing the network policy, there's really nothing that can be done that doesn't compromise the security of the clusters. We will have to communicate this change to customers since anyone using custom Machine Configs will likely be impacted in some way or another.
+
+### Version Skew Strategy
+
+## Implementation History
+
+## Drawbacks
+
+These have been enumerated in other parts of the proposal, but to get them all in one place:
+
+  - This will negatively impact the scale-up time of clusters which make use of post-node-approved Machine Configs.
+  - This has the potential to break a cluster's ability to scale up.
+
+## Alternatives
+
+  - Do nothing and just remove the network policy. This would expose the contents of (what would be) post-node-approved Machine Configs to everything that has network access to the control plane.
+  - Add support for an authentication token to the Machine Config Server (https://github.com/openshift/enhancements/pull/443). The real downside to this approach is that it makes troubleshooting more difficult, as it would likely need to happen in the initramfs of RHCOS. This is an extremely limited environment and requires console or serial access, which isn't available in a number of environments.
```

Comment:
is the experience different here versus when a user provides an invalid pull-secret at install?

Reply:
What is the difference between screwing up 

`https://some-ip/some/value/ignition/master

and 

`https://some-ip/some/value/ignition/master?token=xxxxxxxxxxxxxx`

You either cut and paste the secret correctly or not.  At some point we also have to verify the remote ip (if you ignore validation errors on some-ip you are going to get MITM'd anyway, so security is not your problem).

The person provisioning a box is ultimately responsible for establishing a verifiable, secure, and verified channel to the thing that will establish the basic trust foundation for the rest of that box.  

Why is troubleshooting the problem?  Is that the only argument against requiring a shared secret between provisioner and provisionee to establish mutual verification?

Reply:
@derekwaynecarr if the user provides an invalid pull-secret at install, the bootstrap machine will spin, trying to pull the release image. Eventually, the install will time out and the installer will collect logs. From the perspective of a single compute node, the two scenarios are indistinguishable, but I would expect an admin to look at more than just that. For example, if they SSH into the bootstrap node and run the recommended `journalctl` invocation, it will be immediately obvious that the pull secret is incorrect.

Reply:
@smarterclayton `https://some-ip/some/value/ignition/master` will never change. `https://some-ip/some/value/ignition/master?token=xxxxxxxxxxxxxx` will change regularly (specifically, the token). The concern around troubleshooting is the fact that it will invariably be required more often when the Ignition URL is being changed periodically. There is a deeper question in there as well though. How will this token be rotated? This is easy in deployments which make use of the Machine API Operator, but everywhere else, this will have to be a manual process. And as we all know, if there is a periodic, manual task required for security, we can just assume it's not going to happen. I truly believe that the auth token will hurt the security of our clusters, all things considered.